olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -775,6 +780,78 @@ client::CancellationToken AuthenticationClient::Impl::SignOut(
   });
 }
 
+client::CancellationToken AuthenticationClient::Impl::IntrospectApp(
+    const std::string& access_token, const IntrospectAppCallback& callback) {
+  if (!settings_.network_request_handler) {
+    ExecuteOrSchedule(settings_.task_scheduler, [callback] {
+      AuthenticationError result({static_cast<int>(http::ErrorCode::IO_ERROR),
+                                  "Cannot introspectt app while offline"});
+      callback(result);
+    });
+    return client::CancellationToken();
+  }
+  std::weak_ptr<http::Network> weak_network(settings_.network_request_handler);
```

Comment:
This is tricky now. The reason why SignIn() was done like this is as it was used by TokenEndpoint and it should be sync. For the IntrospectApp this is not the case anymore so I would make the entire call async, not only the response triggering. But this needs to be tested properly.